### DISCUSSION TITLE
Data Importer: "Error accessing "10.0.2.2:8080/api/v1/about?". Status code is 400"

### DISCUSSION THREAD

**Discussion by Adrien-Luxey** [firefly-iii/firefly-iii#6528] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Related to https://github.com/firefly-iii/firefly-iii/issues/3198

Firefly III Data Importer (FIDI) is configured as a Docker container lying next to the Firefly III (Firefly) container. 

When I go the FIDI's homepage, I get this:

![image](https://user-images.githubusercontent.com/6992849/194754962-1c24b22a-9335-40d1-a6ef-5fb5d55a487a.png)

> The importer could not connect to Firefly III. Please remedy the error below first, and check out the[ documentation](https://docs.firefly-iii.org/) if necessary.
> 
> > Error accessing "10.0.2.2:8080/api/v1/about?". Status code is 400. Body is: `<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN"> <html><head> <title>400 Bad Request</title> </head><body> <h1>Bad Request</h1> <p>Your browser sent a request that this server could not understand.<br /> </p> </body></html>`
> 

### Expected behaviour

FIDI should work!

### Debug information

I am on NixOS, and use Podman with the following network option: `--network=slirp4netns:allow_host_loopback=true`.
It allows reaching the host at `10.0.2.2`, useful to reach Posgres on the host at `10.0.2.2:5432` and Firefly (from FIDI) at `10.0.2.2:8080`. It's not related to this, to the issue, as the packets come and go as they should.

Firefly v5.7.12 & FIDI v0.9.16. 

### Steps to reproduce

I'm short on time, sorry. Maybe later, but read on, I think I know the issue.

### Additional info

I setup `tcpdump` in the same network as Firefly, and I think I found the solution.
I think Firefly interprets the line feeds following the `Authorization` header. The header should be trimmed before it is processed.

See this file (my comments start with `#` and are the only ones doing so):

[bug.txt](https://github.com/firefly-iii/firefly-iii/files/9741463/bug.txt)

Gotta fly, ping me if you need additional detail!

**Comment by JC5** [comment]
How are you authenticating to Firefly III?

**Reply by Adrien-Luxey** [reply]
`FIREFLY_III_ACCESS_TOKEN` passed as an environment (`--env`) parameter to `podman run` (really, substitute Podman to Docker in your head, to anything related to spinning containers they behave the same).
I got the token from Firefly and put it inside the environment variable.
And forgot to anonymize it! I will have to change that... (Done)

**Reply by Adrien-Luxey** [reply]
More precisely (if you want to know where the line feed comes from), the token is sourced from a secret file:

```nix
let 
  # Here, we source the token from a file (it ends with a line feed)
  secrets.data_importer_access_token = builtins.readFile /home/firefly/secrets/data_importer_access_token;
in {
  virtualisation.oci-containers.containers.firefly-data-importer = {
    # https://hub.docker.com/layers/fireflyiii/data-importer/version-0.9.16/images/sha256-cd594510734c89e78c64083bfef234c38d20c5d2b8278a2ef5fd4ee9dcd86c94?context=explore
    image = "fireflyiii/data-importer@sha256:60d28f48839ce420b4333335204a0cbfd858634b5e505db04e758d7dbd335aa7";
    environment = {
      FIREFLY_III_ACCESS_TOKEN = secrets.data_importer_access_token;
      FIREFLY_III_URL = "10.0.2.2:8080"; # Host loopback IP when using slirp4netns:allow_host_loopback
      #VANITY_URL = "https://tld";
      TRUSTED_PROXIES = "**";
    };
    ports = [ "127.0.0.1:8081:8080" ];
    extraOptions = [ "--network=slirp4netns:allow_host_loopback=true" ];
  };
}

**Reply by JC5** [reply]
So, the source file contains the newline?

**Comment by Adrien-Luxey** [comment]
I'll verify without, but not 3 line feeds, only one! 

Le 9 octobre 2022 16:02:21 GMT+02:00, James Cole ***@***.***> a écrit :
>So, the source file contains the newline?
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/orgs/firefly-iii/discussions/6528#discussioncomment-3833277
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Envoyé de mon appareil Android avec Courriel K-9 Mail. Veuillez excuser ma brièveté.

**Reply by JC5** [reply]
A quick test with curl gives me no troubles, no matter how many newlines I inject.
